lld/COFF/Writer.cpp

```diff
@@ -1069,6 +1069,16 @@ void Writer::createMiscChunks() {
     debugInfoSec->addChunk(r.second);
   }
 
+  // If .buildid section was not chosen and /build-id is given, create .buildid
+  // section.
+  if (config->buildID && debugInfoSec != buildidSec) {
+    buildidSec->addChunk(debugDirectory);
```

Comment:
Hmm, so we'd add the same `Chunk` to two separate `OutputSection`s? Won't that make a mess for things that try to e.g. layout the sections/chunks, when we assign two conflicting addresses for it in two places?

Reply:
Updated to create new chunks for extra .buildid section.